src/api/altorenderer.cpp

```diff
@@ -0,0 +1,255 @@
+ * File:        altorenderer.cpp
+ * Description: ALTO rendering interface
+ * Author:      Jake Sebright
+ *
+ * (C) Copyright 2018
+ ** Licensed under the Apache License, Version 2.0 (the "License");
+ ** you may not use this file except in compliance with the License.
+ ** You may obtain a copy of the License at
+ ** http://www.apache.org/licenses/LICENSE-2.0
+ ** Unless required by applicable law or agreed to in writing, software
+ ** distributed under the License is distributed on an "AS IS" BASIS,
+ ** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ** See the License for the specific language governing permissions and
+ ** limitations under the License.
+ *
+ **********************************************************************/
+
+#include "baseapi.h"
+#include "renderer.h"
+
+namespace tesseract {
+
+/**********************************************************************
+ * Alto Text Renderer interface implementation
+ **********************************************************************/
+    TessAltoRenderer::TessAltoRenderer(const char *outputbase)
+            : TessResultRenderer(outputbase, "xml") {
+    }
+
+    /**
+    * Append the ALTO XML for the beginning of the document
+    */
+    bool TessAltoRenderer::BeginDocumentHandler() {
+        AppendString(
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                "<alto xmlns=\"http://www.loc.gov/standards/alto/ns-v3#\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:schemaLocation=\"http://www.loc.gov/standards/alto/ns-v3# http://www.loc.gov/alto/v3/alto-3-0.xsd\">\n"
+                "\t<Description>\n"
+                "\t\t<MeasurementUnit>pixel</MeasurementUnit>\n"
+                "\t\t<sourceImageInformation>\n"
+                "\t\t\t<fileName>");
+
+        AppendString(title());
+
+        AppendString("\t\t\t</fileName>\n"
+                     "\t\t</sourceImageInformation>\n"
+                     "\t\t<OCRProcessing ID=\"OCR_0\">\n"
+                     "\t\t\t<ocrProcessingStep>\n"
+                     "\t\t\t\t<processingSoftware>\n"
+                     "\t\t\t\t\t<softwareName>tesseract 4.0.0</softwareName>\n"
+                     "\t\t\t\t</processingSoftware>\n"
+                     "\t\t\t</ocrProcessingStep>\n"
+                     "\t\t</OCRProcessing>\n"
+                     "\t</Description>\n"
+                     "\t<Layout>\n");
+
+        return true;
+    }
+
+    /**
+    * Append the ALTO XML for the end of the document
+    */
+    bool TessAltoRenderer::EndDocumentHandler() {
+        AppendString("\t</Layout>\n</alto>\n");
+
+        return true;
+    }
+
+    /**
+    * Append the ALTO XML for the layout of the image
+    */
+    bool TessAltoRenderer::AddImageHandler(TessBaseAPI *api) {
+        const std::unique_ptr<const char[]> hocr(api->GetAltoText(imagenum()));
```

Comment:
That requires `#include <memory.h>` (see build failures from continuous integration).

Reply:
I'm sorry, my comment was wrong. It should be just `#include <memory>`.